temporalio/worker/replayer.py

```diff
@@ -246,159 +286,26 @@ class ReplayerConfig(TypedDict, total=False):
     build_id: Optional[str]
     identity: Optional[str]
     debug_mode: bool
-    fail_fast: bool
 
 
-@dataclass
-class WorkflowHistory:
-    """A workflow's ID and history."""
+@dataclass(frozen=True)
+class WorkflowReplayResult:
+    """Single workflow replay result."""
 
-    workflow_id: str
-    events: Sequence[temporalio.api.history.v1.HistoryEvent]
+    history: temporalio.client.WorkflowHistory
+    """History originally passed for this workflow replay."""
 
-    @classmethod
-    def from_json(
-        cls, workflow_id: str, history: Union[str, Dict[str, Any]]
-    ) -> WorkflowHistory:
-        """Construct a WorkflowHistory from an ID and a json dump of history.
+    replay_failure: Optional[Exception]
+    """Failure during replay if any.
+    
+    This is not to be confused with a workflow failure, rather this is a task
+    failure such as :py:class:`temporalio.workflow.NondeterminismError`.
```

Comment:
This could easily be interpreted as "nothing is wrong with your workflow". Something like this probably makes more sense:
```suggestion
    This does not mean your workflow exited by raising an error, but
    rather that some task failure such as :py:class:`temporalio.workflow.NondeterminismError`
    was encountered during replay - likely indicating your workflow code
    is incompatible with the history.
```

Reply:
Updated